### PR TITLE
[refactor] Delete @pipeline, execute_pipeline, execute_pipeline_iterator

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Delete `@pipeline`, `execute_pipeline`, and `execute_pipeline_iterator` (`dagster._legacy` functions).

Vast majority of references were removed downstack.

Internal companion PR: https://github.com/dagster-io/internal/pull/5353

## How I Tested These Changes

Existing test suite.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                          * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                            * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                              * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                                * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                                  * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                                    * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                                      * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a> 
                                        * **PR #13311** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13311" target="_blank">Graphite</a> 
                                          * **PR #13312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13312" target="_blank">Graphite</a> 
                                            * **PR #13316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13316" target="_blank">Graphite</a> 
                                              * **PR #13317** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13317" target="_blank">Graphite</a> 
                                                * **PR #13318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13318" target="_blank">Graphite</a> 
                                                  * **PR #13320** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13320" target="_blank">Graphite</a>  👈
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13320?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                          * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                            * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                              * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                                * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                                  * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                                    * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                                      * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a> 
                                        * **PR #13311** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13311" target="_blank">Graphite</a> 
                                          * **PR #13312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13312" target="_blank">Graphite</a> 
                                            * **PR #13316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13316" target="_blank">Graphite</a> 
                                              * **PR #13317** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13317" target="_blank">Graphite</a> 
                                                * **PR #13318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13318" target="_blank">Graphite</a> 
                                                  * **PR #13320** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13320" target="_blank">Graphite</a>  👈
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13320?utm_source=stack-comment).

File: examples/docs_snippets/docs_snippets_tests/concepts_tests/assets_tests/test_materialization_io_managers.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import pandas as pd
 
 from dagster import io_manager, op
-from dagster._legacy import ModeDefinition, execute_pipeline, pipeline
+from dagster._core.definitions.decorators.job_decorator import job
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets_tests/concepts_tests/assets_tests/test_materialization_io_managers.py:4]: nit: make this a dagster import

File: python_modules/dagster/dagster/_core/telemetry.py [hunk 2]
```diff
@@ -75,7 +75,7 @@
 
 
 TELEMETRY_WHITELISTED_FUNCTIONS = {
-    "_logged_execute_pipeline",
+    "_logged_execute_job",
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_core/telemetry.py:78]: I think this is fine - but mind making a change in the telemetry wiki that notes that `execute_pipeline_started` has been renamed?

File: python_modules/dagster/dagster/_legacy/__init__.py [hunk 3]
```diff
@@ -8,22 +8,12 @@
     PresetDefinition as PresetDefinition,
     build_assets_job as build_assets_job,
     default_executors as default_executors,
-    pipeline as pipeline,
 )
 from dagster._core.execution.api import (
-    execute_pipeline as execute_pipeline,
-    execute_pipeline_iterator as execute_pipeline_iterator,
-    reexecute_pipeline as reexecute_pipeline,
+    _reexecute_job as _reexecute_job,
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_legacy/__init__.py:13]: Does this API exist? The pythonic way to perform re-execution should be ReexecutionOptions on `execute_job` now

File: python_modules/dagster/dagster/_utils/test/__init__.py [hunk 4]
```diff
@@ -50,7 +52,7 @@
 )
 
 if TYPE_CHECKING:
-    from dagster._core.execution.results import CompositeSolidExecutionResult, OpExecutionResult
+    pass
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_utils/test/__init__.py:55]: delete

**Review by dpeng817 — APPROVED** [review]
a few nits and one comment worth addressing

File: python_modules/dagster/dagster/_legacy/__init__.py (hunk 3 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_legacy/__init__.py:13]: yeah that ended up there through some fluke, deleted

**Pull request merged**